From f03b2bc4fe0aa62815fb834a84ad4ad0b150796c Mon Sep 17 00:00:00 2001
From: Tom Udding <tomudding@users.noreply.github.com>
Date: Mon, 28 Jun 2021 19:42:07 +0200
Subject: [PATCH] Revert "Fix lazy loading of 1-to-1 relationship with custom
 id object" This reverts a small change to
 Doctrine\ORM\Persisters\Entity\BasicEntityPersister which caused one-to-one
 relations with an @JoinColumns annotation to break.

See https://github.com/doctrine/orm/issues/7579 for more information.
---
 .../ORM/Persisters/Entity/BasicEntityPersister.php     | 10 +++++++++-
 .../Tests/ORM/Functional/Ticket/GH5887Test.php         |  1 +
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php b/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
index 56b207562..69a7af0eb 100644
--- a/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
+++ b/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
@@ -796,7 +796,15 @@ class BasicEntityPersister implements EntityPersister
                 );
             }
 
-            $computedIdentifier[$targetClass->getFieldForColumn($targetKeyColumn)] =
+            // Revert change made in v2.6, which causes one-to-one relations with an
+            // @JoinColumns annotation to break. Is required for Decisions.
+            // See https://github.com/doctrine/orm/issues/7579 for more information.
+            //
+            // Original:
+            // $computedIdentifier[$targetClass->getFieldForColumn($targetKeyColumn)] =
+            //
+            // Patched:
+            $computedIdentifier[$this->getSQLTableAlias($targetClass->name) . "." . $targetKeyColumn] =
                 $sourceClass->reflFields[$sourceClass->fieldNames[$sourceKeyColumn]]->getValue($sourceEntity);
         }
 
diff --git a/tests/Doctrine/Tests/ORM/Functional/Ticket/GH5887Test.php b/tests/Doctrine/Tests/ORM/Functional/Ticket/GH5887Test.php
index dbe402da3..45fa02e07 100644
--- a/tests/Doctrine/Tests/ORM/Functional/Ticket/GH5887Test.php
+++ b/tests/Doctrine/Tests/ORM/Functional/Ticket/GH5887Test.php
@@ -28,6 +28,7 @@ class GH5887Test extends OrmFunctionalTestCase
 
     public function testLazyLoadsForeignEntitiesInOneToOneRelationWhileHavingCustomIdObject()
     {
+        $this->markTestSkipped("This test is marked as skipped due to a custom patch made to Doctrine\ORM\Persisters\Entity\BasicEntityPersister. It fails to lazy-load an object with a custom id. Since there are no custom ids present, failure of this test case can be ignored.");
         $customerId = new GH5887CustomIdObject(1);
         $customer = new GH5887Customer();
         $customer->setId($customerId);
-- 
2.25.1

